futures-io/src/lib.rs

```diff
@@ -148,6 +148,9 @@ mod if_std {
         /// `Interrupted`.  Implementations must convert `WouldBlock` into
         /// `Poll::Pending` and either internally retry or convert
         /// `Interrupted` into another error kind.
+        ///
+        /// `poll_write` must try to make progress by flushing the underlying object before returning
```

Comment:
This bit isn't quite right-- it mustn't flush unless flushing is what is necessary for the underlying IO object to become writable once more.

Reply:
@cramertj sorry to let this linger. I tried asking some clarification, but then the review feature of github kicked in and the comment wasn't actually posted. So trying to resolve this, I could make it say something like:
```rust
/// `poll_write` must try to make progress by flushing the underlying object if that is the only way the underlying object can become writable again.
```
But I feel that is rather confusing. Does that mean it can return pending without trying to flush? Does returning Pending not mean that it is not writable right now and needs to be flushed? What does client code do if `poll_write` returns pending? Do they manually call flush or do they just keep calling `poll_write`? For how long?

Reply:
sorry, that last part doesn't make sense. Obviously if poll_write returns pending, client code waits to be woken up by it. Still not sure what the best formulation is.

Reply:
The example that you wrote in the codeblock above sounds reasonable to me!